### DISCUSSION TITLE
Cropping display in SuperHighRes modes

### DISCUSSION THREAD

**Discussion by ghost** [BlitterStudio/amiberry#1360] [Q&A] [answered]
I want to use Xtreme SuperHighRes Laced AGA mode (1280×1024) in Workbench. My goal is to have pixel exact scaling and output to the host window. For the start, I have set _Scaling Method_ to _Pixelated_ and left the rest at defaults. It was already a huge success, as the display was pixel exact at the first try. My main reason to try AmiBerry was that FS-UAE cannot properly display SuperHighRes modes at all.

There are some minor things however, which I cannot adjust:
- Thick black strip on top of emulator display.
- 8-pixel wide background color vertical strip on the right.
- Background color on the bottom.

I would like to crop them all, retaining only the screen contents. Unfortunately either checking _Manual Crop_ or unchecking _Auto Crop_ ruins the display, looks like it is zoomed in at the center. I've tried to enter "1280" to _Width_ but still image is zoomed in, so I can see only the middle of Amiga screen. BTW I can uncheck both _Manual_ and _Auto_, not sure what does it mean.

My another try was to read the Wiki. I've learned that there are 4 scaling modes, one of them is _Integer_. I expected it won't scale the display when I resize the window, but simply crop the output. It sounded like a solution for me, but this mode is not present in the _Scaling Method_ dropdown list for some reason, changing manually scaling mode to 3 in the config file does not work either.

My question is how can I properly crop the display?

**Comment by midwan** [comment]
Integer scaling is not yet in any of the released versions, but it's in the repository. So, you can only get if for now, if you compile it from source (both `master` and `preview` branches have it). Since I'm working on some bugs still, keep in mind that things might change before the next release.

You didn't mention which version you're testing with. Is it 5.x or 6.x?

The 6.x versions have more accurate compatibility, since they are synced with the latest WinUAE (currently 5.3.0), whereas v5.x is based on WinUAE 4.4.0. The reason for the two versions, is that the newer one is a bit heavier and requires a faster device, while it only has JIT for x86_64 and not ARM.

**Reply by ghost** [reply]
I've tried both versions (to be exact, 6.3.2 and 5.6.8) and described behaviour was the same for both. I run them on x64 machine with XUbuntu 22 installed.

**Reply by midwan** [reply]
OK, trying to recreate this here. Did you need some special monitor drivers to use these modes, if I remember correctly? Do you have a link to those please?

**Reply by ghost** [reply]
Yes, these drivers are not distributed with AmigaOS and may be downloaded [from Aminet](https://aminet.net/package/driver/moni/HighGFXnmore). Installation is standard, one puts driver files with their icons in `DEVS:Monitors`, then restarts the Amiga.

**Reply by midwan** [reply, marked as answer]
Testing this here with the latest from `preview` branch, and I don't see any glitches using that mode. The best results are shown when enabling Integer scaling (for pixel-perfect scaling without any distortion) in full-window mode, as per the wiki: https://github.com/BlitterStudio/amiberry/wiki/Howto:-Enable-Integer-Scaling

If you want to test it, you can try grabbing the latest sources and compiling it, otherwise you can wait a bit until the next release.

**Reply by ghost** [reply]
I've just tested 6.3.3-preview and Integer Scaling works great. The only minor thing left is that Auto Crop grabs a border around the screen as well. It can be removed by resizing the window by hand. I guess there is no way to fix dimensions of the output window. For now I have made some dirty trick using a bash script:
```
./amiberry --conf ./conf/myconf.uae -G &
sleep 15
wmctrl -F -r Amiberry -e 0,0,0,1280,960
```
I first start Amiberry with my config, without displaying UI (in the background, so the script continues). Then wait for switching to Workbench graphics mode (15 seconds is chosen experimentally). Finally `wmctrl` resizes the window.
